UI/DamageIndicator.cs

```diff
@@ -0,0 +1,39 @@
+using Godot;
+
+
+public class DamageIndicator : TextureRect
+{
+	Vector2 ShotFirePosition;
+	float OriginalRotation;
+	public override void _Ready()
+	{
+		// Start transparency Tween in _Ready so we know the Tween has been created
+		Tween TransparencyTween = GetNode<Tween>("Tween");
+		// Tween the "modulate" property
+		TransparencyTween.InterpolateProperty(this, "modulate", new Color(255, 255, 255, 1), new Color(255, 255, 255, 0), 3, Tween.TransitionType.Linear, Tween.EaseType.InOut);
+		TransparencyTween.Start();
+	}
+
+
+	public void SetShotPosition(Vector3 ShotFireArg)
+	{
+		// Flatted the ShotFirePosition
+		ShotFirePosition = new Vector2(ShotFireArg.x, ShotFireArg.z);
+		OriginalRotation = Game.PossessedPlayer.Rotation.y;
+	}
+
+
+	public override void _Process(float delta)
+	{
+		// Calculate our position on a 2D plane
+		Vector2 OurPoint = new Vector2(Game.PossessedPlayer.Translation.x, Game.PossessedPlayer.Translation.z);
+		RectRotation = Mathf.Rad2Deg(OurPoint.AngleToPoint(ShotFirePosition) + OriginalRotation);
```

Comment:
When ShotFirePosition is a Vector3 we can flatten the player pos inline

Reply:
Why do we need ShotFirePosition as a Vector3? We're only using FlatShotFirePos in this class. We never do anything in the 3d world, so I don't think we need to store ShotFirePosition as a Vector3

Reply:
It simplifies accessing of the values of individual axis as Y is actually Y and by using a Vector3 it better demonstrates intent.

Reply:
That makes sense!